charts/cluster-autoscaler/README.md

```diff
@@ -421,6 +421,7 @@ vpa:
 | clusterAPIMode | string | `"incluster-incluster"` | Cluster API mode, see https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/README.md#connecting-cluster-autoscaler-to-cluster-api-management-and-workload-clusters Syntax: workloadClusterMode-ManagementClusterMode for `kubeconfig-kubeconfig`, `incluster-kubeconfig` and `single-kubeconfig` you always must mount the external kubeconfig using either `extraVolumeSecrets` or `extraMounts` and `extraVolumes` if you dont set `clusterAPIKubeconfigSecret`and thus use an in-cluster config or want to use a non capi generated kubeconfig you must do so for the workload kubeconfig as well |
 | clusterAPIWorkloadKubeconfigPath | string | `"/etc/kubernetes/value"` | Path to kubeconfig for connecting to Cluster API managed workloadcluster, only used if `clusterAPIMode=kubeconfig-kubeconfig or kubeconfig-incluster` |
 | containerSecurityContext | object | `{}` | [Security context for container](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) |
+| customArgs | list | `[]` | Additional custom container arguments. Refer to https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-are-the-parameters-to-ca for the full list of cluster autoscaler parameters and their default values. List of arguments as strings. |
```

Comment:
So the idea is to have a sort of backdoor to pass any `--foo=bar` configs that the helm chart doesn't yet have native support for?

Reply:
Yes.
But the main goal is to be able to inject a set of arguments coming from secrets or config maps through environment variables mounted from those secrets/config maps.

Reply:
I don't fully understand this use-case end-to-end. How are we doing this without helm today?

Reply:
Say you mount a env variable called EXTRA_ARGS  from a secret, containing multiple args and values.
You can provide $EXTRA_ARGS to cluster autoscaler command.

Reply:
I tested this scenario and it works fine.

Reply:
Thanks for the explanation, that helps.

I think it would be great to document this use case in greater detail detail as the motivator behind this new helm value. Otherwise folks might be confused how to use it, and start using it for literal key=val options (which may be fine in a sense, but might disincentivize us from maintaining the helm chart values as the safest way to declare configuration).

Does that make sense?

Reply:
Makes sense. What is the best place to document it as short as possible ?

Reply:
I think a Heading 3 (three `#`) before the "GCE" section in `charts/cluster-autoscaler/README.md.gotmpl` to explain the `customArgs` value would be great.

Reply:
Added the doc. Please tell me if this is enough.

Reply:
OOPS. fixed now

Reply:
Thanks, @nicolas-laduguie, for this PR, seems good to me also
/lgtm

Reply:
@orangenagy this is already covered by https://github.com/kubernetes/autoscaler/blob/59aefbcd5e95c90530559038ae1d58fdde5381ea/charts/cluster-autoscaler/templates/deployment.yaml#L130C27-L130C40 providing "_0" extra arguments names.
Example here https://github.com/kubernetes/autoscaler/blob/59aefbcd5e95c90530559038ae1d58fdde5381ea/charts/cluster-autoscaler/values.yaml#L196